pkg/utils/errors/errors.go

```diff
@@ -98,9 +98,13 @@ func (t *reconciliationError) ErrorID() string {
 	return t.errorID
 }
 
+func (t *reconciliationError) Unwrap() error {
+	return t.error
+}
+
 // Cause implements the causer interface and returns the underlying error
 func (t *reconciliationError) Cause() error {
```

Comment:
Can you open an issue for eliminating this legacy `causer` interface to get rid of the confusion?

Reply:
https://github.com/gardener/gardener/issues/6211